### ISSUE TITLE
Background image isn't centered when external monitor connected

### ISSUE THREAD

**Issue by tomasz-oponowicz** [prikhi/lightdm-mini-greeter#19]
## Steps

1. Given an external monitor is connected
1. Given a background image is defined
1. Then lock a screen.

## Expected 

The background image is centered.

## Actual

The top left corner of the background image is in a center of a screen. Screenshot:

![screenshot](https://user-images.githubusercontent.com/615940/36252847-41f7594e-1246-11e8-9ba3-0ca8059b4ebd.jpg)

Enviorment:
```
$ pacman -Q lightdm-mini-greeter 
lightdm-mini-greeter 0.3.2-1
$ pacman -Q lightdm
lightdm 1:1.24.0-3
$ pacman -Q light-locker
light-locker 1.8.0-1
```

**Comment by prikhi** [comment]
Thanks for the report - what's your monitor layout & resolution?

**Comment by ASdeWekker** [comment]
I have the same issue, I've got two 1080p monitors, my wallpaper's resolution is 1920x1080. Hope this helps!

**Comment by tomasz-oponowicz** [comment]
@prikhi 

> I've got two 1080p monitors, my wallpaper's resolution is 1920x1080.

The same.

My `xrandr` output:

```
$ xrandr
Screen 0: minimum 8 x 8, current 1920 x 1080, maximum 32767 x 32767
eDP1 connected (normal left inverted right x axis y axis)
   1920x1080     60.00 +  59.93  
   1680x1050     59.88  
   1400x1050     59.98  
   1600x900      60.00    59.95    59.82  
   1280x1024     60.02  
   1400x900      59.96    59.88  
   1280x960      60.00  
   1368x768      60.00    59.88    59.85  
   1280x800      59.81    59.91  
   1280x720      59.86    60.00    59.74  
   1024x768      60.00  
   1024x576      60.00    59.90    59.82  
   960x540       60.00    59.63    59.82  
   800x600       60.32    56.25  
   864x486       60.00    59.92    59.57  
   640x480       59.94  
   720x405       59.51    60.00    58.99  
   640x360       59.84    59.32    60.00  
DP1 disconnected (normal left inverted right x axis y axis)
DP2 disconnected (normal left inverted right x axis y axis)
HDMI1 disconnected (normal left inverted right x axis y axis)
HDMI2 connected primary 1920x1080+0+0 (normal left inverted right x axis y axis) 480mm x 270mm
   1920x1080     60.00*+  50.00    59.94  
   1920x1080i    60.00    50.00    59.94  
   1600x900      60.00  
   1280x1024     75.02    60.02  
   1152x864      75.00  
   1280x720      60.00    50.00    59.94  
   1024x768      75.03    60.00  
   800x600       75.00    60.32  
   720x576       50.00  
   720x576i      50.00  
   720x480       60.00    59.94  
   720x480i      60.00    59.94  
   640x480       75.00    60.00    59.94  
   720x400       70.08  
VIRTUAL1 disconnected (normal left inverted right x axis y axis)
```

Please let me know if you need more information or output of any specific command.

**Comment by tomasz-oponowicz** [comment]
I mirror the display.

**Comment by tomasz-oponowicz** [comment]
I've just noticed that a layout doesn't matter. I enforced built-in display for a lock screen by modifying `/etc/lightdm/lightdm.conf`:

     display-setup-script=xrandr --output eDP1

...an external display is connected but it is blank. The issue still exists.

However when I disconnect the external display it works as expected.

I hope it helps!

**Comment by ASdeWekker** [comment]
I'm running it on my pc, extending it on my two monitors. Not able to run xrandr right now.

**Comment by prikhi** [comment]
I've got the background image appearing on the correct monitor, still have to center it correctly & deal w/ the background color if the image is smaller than the monitor resolution.

**Comment by davama** [comment]
to expand on @tomasz-oponowicz  suggestion...

You need to turn HDMI2 back on so:

```
 [Seat:*]
session-setup-script=/home/dvmacias/.xmonad/bin/lightdm-sessions.sh
```

where:
```
cat .xmonad/bin/lightdm-sessions.sh 
#!/bin/bash


sleep 3

xrandr --output HDMI2 --auto

sleep 2

xrandr --output HDMI2 --right-of HDMI1
```

Of course it adds a few seconds but im ok with that.

Now if this was fixed in the master then all these edits wouldnt be needed :)
Hope it's useful to someone

**Comment by prikhi** [comment]
I've revisited this.

Started by refactoring the single big background window into a different window for every monitor. That fixes the image positioning & with a little CSS tweak I got the background color to show if the image is smaller than the monitor resolution.

However, this puts the image on every monitor instead of the primary monitor. So what I'll probably do is push the fix, release a new minor version and then make an issue for configuring what monitors the image appears on. Was thinking either `primary`, `others`, `all`?

**Comment by prikhi** [comment]
Should be fixed in 2e577c324671843f6ac28c2c217affb4b94e7541.

If you have time, please test it out. I might just push out a new version now or I might wait to merge some of the PRs.

I don't use the background image stuffs yet(I want to randomize them first), so if you have any other problems, desires, or suggestions, throw them in an issue. Maybe different scaling/fit methods & choosing the monitors to put it on?
